public/js/updateBook.ts

```diff
@@ -9,52 +9,482 @@ interface Book {
+                        authorInput.value = bookData[0].authors[index];
+                    }
+                    else {
+                        authorInput.value = bookData[0].authors[index];
+                    }
+                }
 
-            const publisherInput = document.getElementById('publisherInput') as HTMLInputElement;         
-            publisherInput.value = bookData[0].publisher;
+                for (let index = 0; index < bookData[0].genres.length; index++) {
+                    if (index > 0) {
+                        createInputGenereBox(index);
+                    }
 
-            const publishedInput = document.getElementById('publishedInput') as HTMLInputElement;         
-            publishedInput.value = bookData[0].publishedYear;
+                    let genreInput = document.getElementById('genreInputValue') as HTMLInputElement;
 
-            const IsbnInput = document.getElementById('IsbnInput') as HTMLInputElement;         
-            IsbnInput.value = bookData[0].isbn;
+                    if (index !== 0) {
+                        genreInput = document.getElementById('genreInputValue' + index) as HTMLInputElement;
+                        genreInput.value = bookData[0].genres[index];
+                    }
+                    else {
+                        genreInput.value = bookData[0].genres[index];
+                    }
+                }
 
-            const coverUrlInput = document.getElementById('coverUrlInput') as HTMLTextAreaElement;         
-            coverUrlInput.value = bookData[0].image;
+                const publisherInput = document.getElementById('publisherInput') as HTMLInputElement;
+                publisherInput.value = bookData[0].publisher;
 
-            const coverUrlImgInput = document.getElementById('coverUrlImgInput') as HTMLImageElement;         
-            coverUrlImgInput.src = bookData[0].image;
-            
-            const abstractInput = document.getElementById('abstractInput') as HTMLTextAreaElement;         
-            abstractInput.value = bookData[0].abstract;
+                const publishedInput = document.getElementById('publishedInput') as HTMLInputElement;
+                publishedInput.value = bookData[0].publishedYear;
 
-        } else {
-            console.error('Book not found:');
+                const coverUrlInput = document.getElementById('coverUrlInput') as HTMLTextAreaElement;
+                coverUrlInput.value = bookData[0].image;
+
+                const coverUrlImgInput = document.getElementById('coverUrlImgInput') as HTMLImageElement;
+                coverUrlImgInput.src = bookData[0].image;
+
+                const abstractInput = document.getElementById('abstractInput') as HTMLTextAreaElement;
+                abstractInput.value = bookData[0].abstract;
+
+            } else {
+                console.error('Book not found:');
```

Comment:
I think there is no point in displaying the error in the console. We need to show an error message to the user. 
Have the look at the existing `showSuccessPopup` function.